src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/withToolbar.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+import {autorun} from 'mobx';
+import toolbarStore from './stores/ToolbarStore';
+
+export default function withToolbar(Component) {
+    return class WithToolbar extends Component {
+        componentWillMount() {
+            if (super.componentWillMount) {
+                super.componentWillMount();
+            }
+
+            this.disposer = autorun(() => {
+                toolbarStore.setItems(this.getToolbar());
```

Comment:
I think I came up with a solution now. I can avoid using inheritance when using the `ref` callback, which gives me an instance of the component, where I can access all the methods and properties of the exact instance :-)